aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -525,6 +524,39 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
   iter.cast_outputs();
 }
 
+template <typename scalar_t>
+void cpu_vflip_memcpy(at::TensorIterator& iter) {
```

Comment:
Write a short comment explaining the idea of the algorithm and a couple torch_checks with the preconditions.